active-rfcs/0001-contribution.md

```diff
@@ -0,0 +1,60 @@
+- Feature Name: Contribution
+- Start Date: 2021-03-01
+- Reference Issues:
+- Implementation PR:
+
+# Summary
+
+A contribution plugin will be developed for tichi, which is mainly responsible for labeling external members' PRs with `contribution` or `first-time-contributor` labels.
+
+# Motivation
+
+We need to distinguish between internal and external contributor PRs, and prioritize help with external contributor PRs, especially first-time contributor PRs, which will allow external contributors to have a good contribution experience.
+
+# Detailed design
+
+## contribution
+
+For external contributors, we will label their PR with `contribution`.
+
+### Definition:
+
+- Label Name: `contribution`
+- Usage: External Contributors
+  - We will determine if it is an external contributor by whether it is an org member or not.
+- Available API: https://github.com/kubernetes/test-infra/blob/9b84ac948f6eb781a318fa431d547e577ee668e4/prow/github/client.go#L67
```

Comment:
If I remember correctly, this API must only judge public members.

Reply:
I don't think so.

Please refer to :
- https://docs.github.com/en/rest/reference/orgs#check-organization-membership-for-a-user
- https://docs.github.com/en/rest/reference/orgs#check-public-organization-membership-for-a-user

Reply:
We use the first one.

Reply:
> We use the first one.

Maybe it depends on the way of authorization. If we visit the URL directly, you will jump to the second API.

Reply:
Sorry, I don't understand you .

Reply:
Without permission to access private members, access `https://api.github.com/orgs/tikv/members/Mini256`,
will be redirected to `https://api.github.com/organizations/41004122/public_members/Mini256`

Reply:
What is the result of bringing authorization?

Reply:
I think the reason for this redirect is because only internal members can see private members.

Reply:
I have tested it on postman using token. It works fine.

<img width="1043" alt="image" src="https://user-images.githubusercontent.com/29879298/109472877-a4569280-7aad-11eb-91d1-d26c4ea8680f.png">

Reply:
Got it!